src/pixel.rs

```diff
@@ -0,0 +1,19 @@
+/// A Pixel made up of a compile-time known number of contiguously stored `T`s.
+///
+/// Usually `T` is a small copiable intrinsic type such as `u8`, `u16` or `f32`.
+pub trait Pixel<T, const N: usize> {
+    /// Converts an owned `Pixel` type to an array of its components.
+    #[allow(clippy::wrong_self_convention)]
+    fn as_components(self) -> [T; N];
```

Comment:
What would be a better name for this method to fit rust's API conventions? `into_components()`?

Reply:
I'll switch to `into_components()` for now I think.